test/unit/modules/transport.js

```diff
@@ -13,119 +13,356 @@
  */
 'use strict';
 
-describe('transport', function () {
+var rewire = require('rewire');
+var sinon = require('sinon');
+var chai = require('chai');
+var expect = require('chai').expect;
 
-	describe('Transport constructor', function () {
+var modulesLoader = require('../../common/modulesLoader');
+var TransportModule = rewire('../../../modules/transport.js');
 
-		describe('library', function () {
+describe('transport', function () {
 
-			it('should assign logger');
+	var dbStub, loggerStub, busStub, schemaStub, networkStub, balancesSequenceStub,
+	    transactionStub, blockStub, peersStub, broadcasterStubRef, transportInstance,
+	    library, __private, defaultScope;
 
-			it('should assign db');
+	var restoreRewiredTopDeps;
 
-			it('should assign bus');
+	beforeEach(function (done) {
+		// Recreate all the stubs and default structures before each test case to make
+		// sure that they are fresh every time; that way each test case can modify
+		// stubs without affecting other test cases.
 
-			it('should assign schema');
+		dbStub = {
+			query: sinon.spy()
+		};
 
-			it('should assign network');
+		loggerStub = {
+			debug: sinon.spy(),
+			error: sinon.spy()
+		};
 
-			it('should assign balancesSequence');
+		busStub = {};
+		schemaStub = {};
+		networkStub = {};
+		balancesSequenceStub = {};
 
-			describe('should assign logic', function () {
+		transactionStub = {
+			attachAssetType: sinon.stub()
+		};
 
-				it('should assign block');
+		blockStub = {};
+		peersStub = {};
 
-				it('should assign transaction');
+		restoreRewiredTopDeps = TransportModule.__set__({
+			Broadcaster: function () {
+				broadcasterStubRef = this;
+			}
+		});
 
-				it('should assign peers');
-			});
+		defaultScope = {
+			logic: {
+				block: blockStub,
+				transaction: transactionStub,
+				peers: peersStub
+			},
+			db: dbStub,
+			logger: loggerStub,
+			bus: busStub,
+			schema: schemaStub,
+			network: networkStub,
+			balancesSequence: balancesSequenceStub,
+			config: {
+				peers: {
+					options: {
+						timeout: 1234
```

Comment:
Assign as const

Reply:
Which ones? I declared those as `var` at the top because I need them to be accessible at the top level outside the `beforeEach` block as well. If I use const, I can't set them dynamically like this.

Reply:
right